src/hooks/useActiveDescendantListFocus/index.js

```diff
@@ -27,113 +21,18 @@ function useActiveDescendantListFocus({
   itemsIds,
   isItemSelectable,
   onItemClick,
+  defaultVisualFocusFirstIndex = false,
   focusedElementRole = ROLES.GROUP,
   isHorizontalList = false,
   useDocumentEventListeners = false,
   isIgnoreSpaceAsItemSelection = false
 }) {
-  const pressKeys = useMemo(
-    () => (isIgnoreSpaceAsItemSelection ? [ENTER_KEY] : [ENTER_KEY, SPACE_KEY]),
-    [isIgnoreSpaceAsItemSelection]
-  );
+  const defaultVisualFocusItemIndex = defaultVisualFocusFirstIndex ? 0 : -1;
   const itemsCount = itemsIds.length;
-  const previousFocusedElementRef = usePrevious(focusedElementRef);
```

Comment:
most of this code just moved to useActiveDesscendantListFocusHooks to sub hooks. where the code is new I will write a comment